core/utils/kagome_db_editor.cpp

```diff
@@ -0,0 +1,217 @@
+#include <chrono>
+#include <fstream>
+#include <thread>
+
+#include <boost/di.hpp>
+#include <soralog/impl/configurator_from_yaml.hpp>
+
+#include "blockchain/impl/storage_util.hpp"
+#include "storage/changes_trie/impl/storage_changes_tracker_impl.hpp"
+#include "storage/leveldb/leveldb.hpp"
+#include "storage/trie/impl/trie_storage_backend_impl.hpp"
+#include "storage/trie/impl/trie_storage_impl.hpp"
+#include "storage/trie/polkadot_trie/polkadot_trie_factory_impl.hpp"
+#include "storage/trie/serialization/polkadot_codec.hpp"
+#include "storage/trie/serialization/trie_serializer_impl.hpp"
+#include "utils/profiler.hpp"
+
+namespace di = boost::di;
+using namespace std::chrono_literals;
+using namespace kagome;
+using namespace storage::trie;
+
+template <class T>
+using sptr = std::shared_ptr<T>;
+
+namespace {
+  std::string embedded_config(R"(
+# ----------------
+sinks:
+  - name: console
+    type: console
+    thread: none
+    color: false
+    latency: 0
+groups:
+  - name: main
+    sink: console
+    level: trace
+    is_fallback: true
+    children:
+      - name: kagome-db-editor
+# ----------------
+)");
+}
+
+class Configurator : public soralog::ConfiguratorFromYAML {
+ public:
+  Configurator() : ConfiguratorFromYAML(embedded_config) {}
+};
+
+enum ArgNum : uint8_t { DB_PATH = 1, STATE_HASH, MODE };
+enum Command : uint8_t { COMPACT, DUMP };
+
+void usage() {
+  std::cout << "\n    Kagome DB Editor\n\n";
+  std::cout << "Usage:\n";
+  std::cout << "    kagome-db-editor <db-path> <root-state> <command>\n\n";
+  std::cout << "    <db-path> full or relative path to kagome database. It is "
+               "usually path "
+               "polkadot/db inside base path set in kagome options.\n";
+  std::cout << "    <root-state> root state hash in 0x prefixed hex format.\n";
+  std::cout << "    <command>\n";
+  std::cout << "        dump: Dumps the state from the DB to file "
+               "hex_full_state.yaml in "
+               "format ready for use in polkadot-test.\n";
+  std::cout << "        compact: Compacts the kagome DB. Leaves only keys of "
+               "the state "
+               "passed as an arguments. Removes all other keys. [Default]\n\n";
+  std::cout << "Example:\n";
+  std::cout << "    kagome-db-editor base-path/polkadot/db 0x1e22e dump\n";
```

Comment:
Strings are so warped. You can use comment 
`// clang-format off`
for switching off formatting and 
`// clang-format on` for switching on it back.

Reply:
I did better, I think